diffpy/snmf/stretchednmfapp.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+import argparse
+
+from diffpy.snmf.io import load_input_signals, initialize_variables
+from diffpy.snmf.subroutines import update_weights_matrix, get_residual_matrix, objective_function
+
+
+def create_parser():
+    parser = argparse.ArgumentParser(
+        prog="stretched_nmf",
+        description="Stretched Nonnegative Matrix Factorization"
+    )
+    parser.add_argument('-v', '--version', action='version', help='Print the software version number')
+    parser.add_argument('-i', '--input-directory', type=str, default=None,
+                        help="Directory containing experimental data. Default before will cause the program to use the current working directory as the input directory.")
+    parser.add_argument('-o', '--output-directory', type=str,
+                        help="The directory where the results will be dumped. Default behavior will create a new directory named 'smnf_results' inside the input directory.")
+    parser.add_argument('-t', '--data-type', type=str, choices=['xrd', 'pdf'],
+                        help="The type of the experimental data.")
+    parser.add_argument('components', type=int,
+                        help="The number of component signals to obtain from experimental "
+                             "data. Must be an integer greater than 0.")
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    args = create_parser()
+
+    grid, data_input, data_type = load_input_signals(args.input_directory)
+    if args.data_type is not None:
+        variables = initialize_variables(data_input, args.components, args.data_type)
+    else:
+        variables = initialize_variables(data_input, args.components, data_type)
+
+    if variables["data_type"] == 'pdf':
```

Comment:
this is important functionality and needs to be in subroutines.  Also, we generally lift with the option of different amounts, so `f.min*amt` sthg like that.  amt (give it a more user-friendly name) could be input by user but defaults to 1.

Reply:
Ok, I will do this. I will make a new function in subroutines.

Should I also add the ability for the user to specify the amount through the cli and io.

Reply:
amount is for something that is continous...amount of sugar, amount of sand.  Let's use "number" instead of amount (for countable things, number of sheep, number of students).   Now, your question is referring to number of what?  If it is needed it will need to be input, possibly as an optional parameter with a good default chose.

Reply:
I understand. I will make the lift factor an optional parameter.